api/src/main/java/marquez/service/models/Job.java

```diff
@@ -60,7 +62,8 @@ public Job(
       @Nullable final ImmutableMap<String, String> context,
       @Nullable final String description,
       @Nullable final Run latestRun,
-      @Nullable final ImmutableMap<String, Object> facets) {
+      @Nullable final ImmutableMap<String, Object> facets,
+      @NonNull Optional<UUID> currentVersionUuid) {
```

Comment:
I would continue to follow the convention in the class and pass the `UUID` without the `Optional` wrapper and define a `getCurrentVersion()` method that returns `Optional<UUID>`

Reply:
what is a reason for doing that? If we have mix of approaches?
cons:
- (more code) I need to add getter manually
- it would create Optional object at every call, instead of once in constructor

Reply:
We will be revising all API models for Marquez core, clients, etc when we [generate models from the OpenAPI spec](https://github.com/MarquezProject/marquez/issues/1472). I know currently `Optional<>` are used differently depending on the class (which we'll want to add guidelines for), but ["Optional" should not be used for parameters](https://rules.sonarsource.com/java/RSPEC-3553). I would conform to what's currently used in the class, but when we generate the models, this will no longer be something we have to write ourselves.

Reply:
I also don't like to have it as a param, I prefer to have UUID value as a param, and then wrap it in constructor. It's not recommended to use Optional as a field also, but it relates to serialization/deserialization which works fine with jackson java8module

Reply:
I've changed it in your way, should be ok